### ISSUE TITLE
Bug - Tables are missing primary keys

### ISSUE THREAD

**Issue by bluikko** [znuny/Znuny#238]
### Environment

- OS: Linux EL7          
- Browser: Chrome     
- Znuny version: 6.0.39

### Expected behaviour
All SQL tables should have PRIMARY KEYs. These are required for RBR and InnoDB cluster systems such as Galera to work reliably.  
This requirement becomes more and more important as users move to hosted/"cloud" solutions where often the SQL service uses such a clustering solution.

UNIQUE KEYs are not enough, they need to be PRIMARY KEYs. Some of the existing KEY (especially UNIQUE KEYs?) possibly could be changed into PRIMARY KEY or a new one could be added.

### Actual behaviour
Several tables are missing PRIMARY KEYs on this system (updated since a very, very old version so perhaps not all of these are used anymore):
```
acl_sync
article_flag
calendar_appointment_ticket
customer_preferences
customer_user_customer
generic_agent_jobs
group_customer
group_customer_user
group_role
group_user
link_relation
notification_event_item
personal_queues
personal_services
pm_entity_sync
postmaster_filter
queue_preferences
queue_standard_template
role_user
search_profile
service_customer_user
service_preferences
service_sla
sla_preferences
ticket_flag
ticket_index
ticket_lock_index
ticket_loop_protection
ticket_watcher
user_preferences
virtual_fs_preferences
web_upload_cache
xml_storage
```

### How to reproduce

1. Install Znuny or update from previous system.
2. Check which tables have PRIMARY KEY.
3. Note several tables do not have PRIMARY KEY.

**Comment by Aquifoliales** [comment]
It seems, that the table
`process_id`
is also affected.

**Comment by NiklasSchmitt** [comment]
internal issue 406

**Comment by NiklasSchmitt** [comment]
This is solved in the next Znuny version 7.1.1. See https://github.com/znuny/Znuny/commit/c508ccc5b7e40109d6ad2af637cc965fec51dbdc for technical details
